code/parse/sexp.cpp

```diff
@@ -10085,43 +10111,48 @@ int test_argument_nodes_for_condition(int n, int condition_node, int *num_true,
 		// only eval this argument if it's valid
 		if (Sexp_nodes[n].flags & SNF_ARGUMENT_VALID)
 		{
-			// evaluate conditional for current argument
-			Sexp_replacement_arguments.emplace_back(Sexp_nodes[n].text, n);
+			// evaluate conditional for current argument(s)
+			const int num_args = copy_node_to_replacement_args(n);
+
 			val = eval_sexp(condition_node);
-			if ( Sexp_nodes[condition_node].value == SEXP_KNOWN_TRUE ||
-					Sexp_nodes[condition_node].value == SEXP_KNOWN_FALSE) {
+			if (Sexp_nodes[condition_node].value == SEXP_KNOWN_TRUE ||
+				Sexp_nodes[condition_node].value == SEXP_KNOWN_FALSE) {
 				val = Sexp_nodes[condition_node].value;
-			} else if ( Sexp_nodes[condition_node].value == SEXP_NAN_FOREVER ) {
+			} else if (Sexp_nodes[condition_node].value == SEXP_NAN_FOREVER) {
 				// In accordance with SEXP_NAN/SEXP_NAN_FOREVER becoming SEXP_FALSE in eval_sexp()
 				val = SEXP_KNOWN_FALSE;
 			}
```

Comment:
This must be within the for loop.
You have to test the validity of the condition for _each_ of the potential arguments, and only apply them to the ``Applicable_arguments_temp`` if that specific argument returns true.
Otherwise, the validity of the first container argument would determine whether or not all container arguments are evaluated, which is evidently incorrect

Reply:
Good catch!

I initially thought it didn't need to be in the loop, since `condition_node` was the only node passed to `eval_sexp()`, but then realized that `<argument>` is going to be different for each call to `eval_sexp()`.

The test mission "Containers as Arguments" that Karajorma posted in #fred on Discord now works correctly again.